src/gleam/list.gleam

```diff
@@ -1565,6 +1565,50 @@ pub fn each(list: List(a), f: fn(a) -> b) -> Nil {
   }
 }
 
+/// A variant of each that might fail.
+///
+/// Takes a function that returns a result and tries to calls it for each element in a list.
```

Comment:
Not sure if my wording is better but it would be this:

> Used to iterate through a list of items and execute a side effect per item as long as the previous side effect execution returned an `Ok(_)`.
> 
> Takes a function that returns a `Result` which is applied to each list item. If a the function application on an item returns `Ok(_)` this function will recurse and apply to the next item in the list. If however, if the application returns `Error(_)` the function will stop executing at that point.
> 
> At any point the function executes it will return `Nil`, even if it passes through all items or errors on any item.

Reply:
Yes I think yours is way better! Thanks 🙏